From 32dea1328dddd8d2b2c0f0015bc05b076c9d1432 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Wed, 20 Mar 2013 13:14:55 +0100
Subject: [PATCH 0987/1083] drm/rcar-du: Fix crash when disabling an already
 disabled plane

The plane disable code tries to dereference a NULL pointer if the plane
is already disabled. Return immediately in that case.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit 1e07c427ebbbe74a5f7279d6603d8bffc1315328)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit f9ec27574094fde6e85fafd2041fdf0b57ada7b2)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_plane.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_plane.c b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
index 15b8ff7..75197ad 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_plane.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_plane.c
@@ -312,6 +312,9 @@ static int rcar_du_plane_disable(struct drm_plane *plane)
 	struct rcar_du_device *rcdu = plane->dev->dev_private;
 	struct rcar_du_plane *rplane = to_rcar_plane(plane);
 
+	if (!rplane->enabled)
+		return 0;
+
 	mutex_lock(&rcdu->planes.lock);
 	rplane->enabled = false;
 	rcar_du_crtc_update_planes(rplane->crtc);
-- 
1.8.3.2

